### ISSUE TITLE
Test smb_docker

### ISSUE THREAD

**Issue by shobhitsinghal1** [ETHZ-RobotX/smb_docker#4]
Test if the docker container works on multiple OSs and systems:
Current branch: feature/open3d

- [x] - Windows (running well on Win 11)
- [x] - Ubuntu (Nvidia/integrated)
- [x] - Mac Intel (still causes issues mainly with opengl)
- [ ] - Mac M1

- [x] - Modify dockerfile to include open3d slam
- [ ]  - Final testing and documentation: to be done after all the packages are moved to master

**Comment by ShobhitSinghal1** [comment]
Windows: Using docker desktop: extended instructions

For open3d installation after running the docker image:
- Upgrade pip: use pip to upgrade but it does not upgrade the system level pip. So `sudo apt purge pip` and then add alias in ~/.bash_aliases as `alias pip=/root/.local/bin/pip3`. Check `pip --version`, should be latest.
- Install pyyaml latest version by ignoring current dist-utils pyyaml `pip -I install pyyaml`
- Install open3d: `pip install open3d==0.15.2`
- Install open3d dependencies for ubuntu: `curl https://raw.githubusercontent.com/isl-org/Open3D/master/util/install_deps_ubuntu.sh | sudo bash -s -- assume-yes`
- Install latest cmake from kitware: follow https://apt.kitware.com/
- sudo apt install liblua5*-dev
- git clone https://github.com/leggedrobotics/open3d_slam.git slam/open3d_slam in src directory
- catkin build open3d_slam_ros

**Comment by ShobhitSinghal1** [comment]
Open3d installs well. I have tested that open3d_slam_ros builds as well but with a temporary smb.repos link and other manual changes as it is not public yet. Once everything is merged into master, this needs to be tested finally once.

**Comment by mantelt** [comment]
that's good news! 

As discussed, maybe @daBaret can assist in adjusting the Docker image that is uploaded to Dockerhub. I will look into all the automated builds as well hopefully soon.

**Comment by mantelt** [comment]
Hi @ShobhitSinghal1 

Quick question: I finally had a look at your [`feature/open3d` branch](https://github.com/ETHZ-RobotX/smb_docker/tree/feature/open3d). Is there a particular reason why you're not "just" installing the open3d ppa such that open3d_slam_ros compiles?

**Comment by mantelt** [comment]
I wanted to get the docker environment up & running again, as well as the automatic builds that depend on the docker, so I added  open3d as a PPA. As soon as the newest changes regarding open3d_slam are pushed to master in smb_common, the new version of  open3d_slam should be in the docker as well by default.

@ETHZ-RobotX/rss-hiwis , let me know if you see any issue with my recent changes.

**Comment by ShobhitSinghal1** [comment]
Hi @mantelt 
Thanks for taking a look. The main reason was that it was a different version on the ppa than the one required. So, I went the pip way and had to update pip to do that. But if it works then no issues, it is much cleaner as well. Also we need to update Open3D_DIR variable, I will add that.

**Comment by mantelt** [comment]
Unfortunately, I didn't have time yet to test the updated docker image. I was mostly trying to fix the compilation issues. This I actually managed by using the PPA for open3d.
